src/plugins/intel_gpu/src/graph/impls/ocl/non_max_suppression.cpp

```diff
@@ -171,11 +173,25 @@ struct non_max_suppression_impl : typed_primitive_impl_ocl<non_max_suppression>
 namespace detail {
 
 attach_non_max_suppression_impl::attach_non_max_suppression_impl() {
-    implementation_map<non_max_suppression>::add(impl_types::ocl, non_max_suppression_impl::create, {
-        std::make_tuple(data_types::i32, format::bfyx),
-        std::make_tuple(data_types::f16, format::bfyx),
-        std::make_tuple(data_types::f32, format::bfyx),
-    });
+    implementation_map<non_max_suppression>::add(impl_types::ocl,
+                                                 non_max_suppression_impl::create,
+                                                 {
+                                                     std::make_tuple(data_types::i32, format::bfyx),
+
+                                                     std::make_tuple(data_types::f16, format::bfyx),
+                                                     std::make_tuple(data_types::f16, format::b_fs_yx_fsv16),
```

Comment:
Overall impl seems good, but since the blocked format might affect the performance impact, we need to check it by ourselves. Please wait for the result.